core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -181,6 +184,8 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
             UnknownQualAppResult(pathStr, app.appId,
               "No aggregated stats for event log")
           }
+        case result =>
+          throw new IllegalArgumentException(s"Unsupported result type: $result")
```

Comment:
This is not a good usage for `IllegalArgumentException`
Usually `IllegalArgumentException` is thrown for function arguments.

I understand that problem here that we catch SuccessResult and FailureResult; but there are many other classes inherit from `OperationResult`
Theoretically, the pattern of success/failure implies that `FailureResult` handles anything that is not a `success`. This implies that there should not be more than two case clauses.

Probably, the class definitions break this pattern.

Reply:
Replace it with `Either` type